### PR TITLE
Upgrading outdated Gradle Wrapper and dependencies.

### PR DESCRIPTION
**Pull request by srayhunter**

### CODE REVIEW

**Comment by sublimator** [comment]
haha, hello again, was just trolling this repo

**Comment by srayhunter** [comment]
@sublimator I want to get some android stuff working and wanted to update gradle and the deps for it.

File: build.gradle [hunk 1]
```diff
@@ -1,8 +1,8 @@
 apply plugin: 'java'
 apply plugin: 'ivy-publish'
 
-sourceCompatibility = 1.6
-targetCompatibility = 1.6
+sourceCompatibility = JavaVersion.VERSION_1_7
```
Comment (sublimator) [inline build.gradle]: Android doesn't support v8 these days? Only the very latest?

Comment (srayhunter) [reply build.gradle]: In my new commit this has been changes to 1.7.

**Comment by sublimator** [comment]
All goodness?

**Comment by srayhunter** [comment]
I am going to create a new branch and fix some more items and then create a new PR.

**Pull request closed without merging**